src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx

```diff
@@ -86,11 +111,125 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
 
     // Continue without filling in required fields
     userEvent.click(screen.getByText(/Continue/i));
 
     expect((await screen.findAllByRole('alert')).length).to.equal(2);
     expect(screen.history.push.called).to.be.false;
   });
+
+  it('should display list of providers when choose a provider clicked', async () => {
+    const store = createTestStore(initialState);
+    await setTypeOfCare(store, /primary care/i);
+    await setTypeOfFacility(store, /Community Care/i);
+    const screen = renderWithStoreAndRouter(
+      <CommunityCareProviderSelectionPage />,
+      {
+        store,
+      },
+    );
+
+    // Trigger provider list loading
+    userEvent.click(await screen.findByText(/Choose a provider/i));
+    expect(await screen.findByText(/your address on file:/i)).to.exist;
+
+    // Verify provider list count and get load more button
+    expect(screen.baseElement).to.contain.text(
+      '123 big sky stCincinnati, OH 45220',
+    );
+    expect(screen.baseElement).to.contain.text(
+      'Displaying 1 to 5 of 16 providers',
+    );
+    expect((await screen.findAllByRole('radio')).length).to.equal(7);
+
+    userEvent.click(await screen.findByText(/\+ 5 more providers/i));
+    expect(await screen.findByText(/displaying 1 to 10 of 16 providers/i)).to
+      .exist;
+    expect((await screen.findAllByRole('radio')).length).to.equal(12);
+
+    userEvent.click(await screen.findByText(/\+ 5 more providers/i));
+    expect(await screen.findByText(/displaying 1 to 15 of 16 providers/i)).to
+      .exist;
+    expect((await screen.findAllByRole('radio')).length).to.equal(17);
+
+    userEvent.click(await screen.findByText(/\+ 1 more providers/i));
+    expect(await screen.findByText(/displaying 1 to 16 of 16 providers/i)).to
+      .exist;
+    expect((await screen.findAllByRole('radio')).length).to.equal(18);
+
+    // Choose Provider
+    userEvent.click(await screen.findByText(/AJADI, ADEDIWURA/i));
+    userEvent.click(
+      await screen.getByRole('button', { name: /choose provider/i }),
+    );
+    expect(screen.baseElement).to.contain.text(
+      'AJADI, ADEDIWURA700 CONSTITUTION AVE NEWASHINGTON, DC 20002-65999349.3 miles',
+    );
+
+    // Change Provider
+    userEvent.click(
+      await screen.findByRole('button', { name: /change provider/i }),
+    );
+    userEvent.click(await screen.findByText(/LYONS, KRISTYN/i));
+    userEvent.click(
+      await screen.findByRole('button', { name: /choose provider/i }),
+    );
+
+    expect(screen.baseElement).to.contain.text(
+      'LYONS, KRISTYN1785 S HAYES STARLINGTON, VA 22202-27149347.1 miles',
+    );
+
+    // Cancel Selection (not clearing of a selected provider)
+    userEvent.click(
+      await screen.findByRole('button', { name: /change provider/i }),
+    );
+    expect(await screen.findByText(/displaying 1 to 5 of 16 providers/i)).to
+      .exist;
+    expect(await screen.findByRole('button', { name: /cancel/i })).to.exist;
```

Comment:
It doesn't seem like you click on the Cancel button here, if that was the intent of the test

Reply:
Good catch, yeah that was the intent. Fixed